src/app/container/container.component.ts

```diff
@@ -197,8 +197,16 @@ export class ContainerComponent extends Entry implements AfterViewInit {
     if (tool) {
       this.tool = tool;
       this.initTool();
+      this.requestAccessHREF$ = this.extendedTool$.pipe(
+        map((data) => {
+          if (data.imgProvider) {
+            return this.emailService.composeRequestAccessEmail(this.tool, data.imgProvider);
```

Comment:
the extendedTool should've been passed in here, not `this.tool`

Reply:
I'm still a bit concerned about the problem with consistency, because the parameter names and their types are all the same for a couple functions in `email.service.ts`. Developers might assume they are all the same but in the end they are not. Or I could change `public composeContactAuthorEmail(tool: ExtendedDockstoreTool): string` in `email.service.ts` - their parameter to be type `DockstoreTool`. That might be clearer? @garyluu

Reply:
It's probably more clear that I just push my code for you to see

Reply:
current changes look fine